llvm/include/llvm/Target/TargetMachine.h

```diff
@@ -420,6 +420,11 @@ class TargetMachine {
   virtual unsigned getAddressSpaceForPseudoSourceKind(unsigned Kind) const {
     return 0;
   }
+
+  /// True if target has some form of pointer arithmetic checking.
+  /// Helps identify whether pointer arithmetic semantics should be preserved
+  /// for passes such as instruction selection.
+  virtual bool isPtrArithmeticChecked(const Function &F) const { return false; }
```

Comment:
Don't see why this needs to be in TargetMachine. It seems to only depend on the subtarget, so can go directly there?

Reply:
Plus the only use is in AArch64